lib/facter/java_version.rb

```diff
@@ -16,11 +16,12 @@
 Facter.add(:java_version) do
   setcode do
     if ['darwin'].include? Facter.value(:kernel).downcase
-      return nil unless Facter::Core::Execution.execute('/usr/libexec/java_home --failfast', { on_fail: false })
-    else
-      return nil unless Facter::Core::Execution.which('java')
+      if Facter::Core::Execution.execute('/usr/libexec/java_home --failfast', { on_fail: false })
+        version = Facter::Core::Execution.execute('java -Xmx12m -version 2>&1').lines.find { |line| line.include?('version') }
+      end
+    elsif Facter::Core::Execution.which('java')
```

Comment:
was this an else statement before? Now you have an if and an elsif... wondering what happens if neither the if nor the elsif conditions are satisfied?

Reply:
@GSPatton  When neither of the elsif conditions are satisfied is what we needed to test for. When both conditions are false, the java binary is not present on the machine, which ensures that the exec statement isn't run on a missing binary (that's what was causing the error). :-)

Reply:
So in other words, the 
```version = Facter::Core::Execution.execute('java -Xmx12m -version 2>&1').lines.find { |line| line.include?('version') } ```
only executes when java is installed on the target machine

Reply:
great stuff. Thanks for the explanation